### ISSUE TITLE
Tensorflow 2.2: No gradients provided for any variable

### ISSUE THREAD

**Issue by locdoan12121997** [tensorflow/tensorflow#41343]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): pip install
- TensorFlow version (use command below): 2.2
- Python version: 3.6.9
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1/ 7.6.5
- GPU model and memory:  GPU Nvidia 1080Ti / 12GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I build a custom Unet with custom Densenet encoder. After I compile model and use fit_generator, I got No gradients provided for any variable

**Describe the expected behavior**
Model run the training

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
Link colab https://drive.google.com/file/d/1fyb88Ako0CjoVP9O5NQdy6rfjoBhXaFS/view?usp=sharing

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by locdoan12121997** [comment]
Hi, it's not a bug, it's how I create the data generator. It is fixed by change 
```return [x1, x2, x3, x4], y```
to
```yield [x1, x2, x3, x4], y```
